src/realm/table.cpp

```diff
@@ -3155,52 +3138,251 @@ void Table::set_string(size_t col_ndx, size_t ndx, StringData value, bool is_def
+        throw LogicError(LogicError::binary_too_big);
+    set_binary_big(col_ndx, ndx, value, is_default);
+}
 
-size_t Table::set_string_unique(size_t col_ndx, size_t ndx, StringData value)
+template<>
+void Table::set(size_t col_ndx, size_t ndx, Mixed value, bool is_default)
 {
-    if (REALM_UNLIKELY(value.size() > max_string_size))
-        throw LogicError(LogicError::string_too_big);
-    if (REALM_UNLIKELY(!is_attached()))
-        throw LogicError(LogicError::detached_accessor);
-    if (REALM_UNLIKELY(ndx >= m_size))
-        throw LogicError(LogicError::row_index_out_of_range);
-    // For a degenerate subtable, `m_cols.size()` is zero, even when it has a
-    // column, however, the previous row index check guarantees that `m_size >
-    // 0`, and since `m_size` is also zero for a degenerate subtable, the table
-    // cannot be degenerate if we got this far.
+    REALM_ASSERT_3(col_ndx, <, get_column_count());
+    REALM_ASSERT_3(ndx, <, m_size);
+    bump_version();
 
-    if (!is_nullable(col_ndx) && value.is_null())
-        throw LogicError(LogicError::column_not_nullable);
+    MixedColumn& col = get_column_mixed(col_ndx);
+    DataType type = value.get_type();
 
-    if (!has_search_index(col_ndx))
-        throw LogicError(LogicError::no_search_index);
+    switch (type) {
+        case type_Int:
+            col.set_int(ndx, value.get_int()); // Throws
+            break;
+        case type_Bool:
+            col.set_bool(ndx, value.get_bool()); // Throws
+            break;
+        case type_OldDateTime:
+            col.set_olddatetime(ndx, value.get_olddatetime()); // Throws
+            break;
+        case type_Timestamp:
+            col.set_timestamp(ndx, value.get_timestamp()); // Throws
+            break;
+        case type_Float:
+            col.set_float(ndx, value.get_float()); // Throws
+            break;
+        case type_Double:
+            col.set_double(ndx, value.get_double()); // Throws
+            break;
+        case type_String:
+            if (REALM_UNLIKELY(value.get_string().size() > max_string_size))
+                throw LogicError(LogicError::string_too_big);
+            col.set_string(ndx, value.get_string()); // Throws
+            break;
+        case type_Binary:
+            if (REALM_UNLIKELY(value.get_binary().size() > ArrayBlob::max_binary_size))
+                throw LogicError(LogicError::binary_too_big);
+            col.set_binary(ndx, value.get_binary()); // Throws
+            break;
+        case type_Table:
+            col.set_subtable(ndx, nullptr); // Throws
+            break;
+        case type_Mixed:
+        case type_Link:
+        case type_LinkList:
+            REALM_ASSERT(false);
+            break;
+    }
+
+    if (Replication* repl = get_repl())
+        repl->set_mixed(this, col_ndx, ndx, value, is_default ? _impl::instr_SetDefault : _impl::instr_Set); // Throws
+}
+
+template<>
+size_t Table::set_unique(size_t col_ndx, size_t row_ndx, null)
```

Comment:
Now you have taken the time to rearrange the functions, I think this should be moved along the other set_unique templates